project/Build.scala

```diff
@@ -419,9 +420,11 @@ object Build {
       ),
 
       // For convenience, change the baseDirectory when running the compiler
-      baseDirectory in (Compile, run) := baseDirectory.value / "..",
+      (Compile / forkOptions) := (Compile / forkOptions).value.withWorkingDirectory((ThisBuild / baseDirectory).value),
+      (Compile / run / forkOptions) := (Compile / run / forkOptions).value.withWorkingDirectory((ThisBuild / baseDirectory).value),
       // And when running the tests
-      baseDirectory in Test := baseDirectory.value / "..",
+      (Test / forkOptions) := (Test / forkOptions).value.withWorkingDirectory((ThisBuild / baseDirectory).value),
+      (Test / testOnly / forkOptions) := (Test / testOnly / forkOptions).value.withWorkingDirectory((ThisBuild / baseDirectory).value),
```

Comment:
This setting is redundant with the one above because `<project> / Test / testOnly / forkOptions` is by default provided by `<project> / Test / forkOptions`:
```shell
sbt:scala3> inspect scala3-compiler / Test / testOnly / forkOptions
[info] Task: sbt.ForkOptions
[info] Description:
[info] 	Configures JVM forking.
[info] Provided by:
[info] 	ProjectRef(uri("file:/home/piquerez/lampepfl/dotty/"), "scala3-compiler") / Test / forkOptions
...
```

The same is not true however for `<project> / Compile / run / forkOptions` and `<project> / Compile / forkOptions>`

Reply:
```suggestion
```